truffle-config.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Use this file to configure your truffle project. It's seeded with some
+ * common settings for different networks and features like migrations,
+ * compilation and testing. Uncomment the ones you need or modify
+ * them to suit your project as necessary.
+ *
+ * More information about configuration can be found at:
+ * 
+ * https://trufflesuite.com/docs/truffle/reference/configuration
+ *
+ * To deploy via Infura you'll need a wallet provider (like @truffle/hdwallet-provider)
+ * to sign your transactions before they're sent to a remote public node. Infura accounts
+ * are available for free at: infura.io/register.
+ *
+ * You'll also need a mnemonic - the twelve word phrase the wallet uses to generate
+ * public/private key pairs. If you're publishing your code to GitHub make sure you load this
+ * phrase from a file you've .gitignored so it doesn't accidentally become public.
+ *
+ */
+const HDWalletProvider = require('@truffle/hdwallet-provider');
+
+//const fs = require('fs');
+//const mnemonic = fs.readFileSync(".secret").toString().trim();
+//const private_keys = fs.readFileSync(".private_keys").toString().trim();
+require('dotenv').config()
+const {RPC_URL, PRIVATE_KEY, ETHERSCAN_API_KEY} = process.env;
```

Comment:
well done. I guess you have `.env` file where you list variables. Let's have `.env.template` file which can be a template for `.env` file and contract deployers can create `.env` file from  `.env.template` and specify correct values for variables.